### DISCUSSION TITLE
400 Error when creating a new issue

### DISCUSSION THREAD

**Discussion by AnotiaWang** [community/community#10470] [Projects and Issues]
Hi! I tried to create an issue yesterday in [this repo](https://github.com/Fndroid/clash_for_windows_pkg/issues), which uses the [issue form template](https://docs.github.com/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms). When I click "Submit", the page returns `HTTP ERROR 400`.

Here are some info that may help:

<details>
<summary>Response Headers:</summary>

```
content-length: 0
content-security-policy: default-src 'none'; base-uri 'self'; block-all-mixed-content; child-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/; connect-src 'self' uploads.github.com objects-origin.githubusercontent.com www.githubstatus.com collector.githubapp.com api.github.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com cdn.optimizely.com logx.optimizely.com/v1/events translator.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com objects-origin.githubusercontent.com; frame-ancestors 'none'; frame-src render.githubusercontent.com viewscreen.githubusercontent.com notebooks.githubusercontent.com; img-src 'self' data: github.githubassets.com media.githubusercontent.com camo.githubusercontent.com identicons.github.com collector.githubapp.com avatars.githubusercontent.com github-cloud.s3.amazonaws.com objects.githubusercontent.com objects-origin.githubusercontent.com secured-user-images.githubusercontent.com/; manifest-src 'self'; media-src github.com user-images.githubusercontent.com/; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com; worker-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/
content-type: text/html; charset=utf-8
date: Mon, 24 Jan 2022 02:29:36 GMT
expect-ct: max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"
referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
server: GitHub.com
strict-transport-security: max-age=31536000; includeSubdomains; preload
vary: Accept-Encoding, Accept, X-Requested-With
x-content-type-options: nosniff
x-frame-options: deny
x-github-request-id: 5CA9:1A84:2C01FAF:33384EB:61EE0EF3
x-xss-protection: 0
```

</details>

<details>
<summary>Request Headers:</summary>

```
:authority: github.com
:method: POST
:path: /Fndroid/clash_for_windows_pkg/issues
:scheme: https
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
accept-encoding: gzip, deflate, br
accept-language: zh-CN,zh;q=0.9,en-US;q=0.8,en;q=0.7
cache-control: no-cache
content-length: 3973
content-type: application/x-www-form-urlencoded
cookie: *****************
dnt: 1
origin: https://github.com
pragma: no-cache
referer: https://github.com/Fndroid/clash_for_windows_pkg/issues/new?assignees=&labels=&template=bug.yaml&title=%5BBug%5D%3A+
sec-ch-ua: " Not;A Brand";v="99", "Google Chrome";v="97", "Chromium";v="97"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Windows"
sec-fetch-dest: document
sec-fetch-mode: navigate
sec-fetch-site: same-origin
sec-fetch-user: ?1
upgrade-insecure-requests: 1
user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36
```

</details>

**Comment by AnotiaWang** [comment]
And there's another bug I found when completing the issue form:

For checkboxes, only the first two can be ticked by clicking the text, clicks on the other boxes will be directed to the second box

Screen recording GIF:

![bandicam 2022-01-24 10-40-29-285 (1)](https://user-images.githubusercontent.com/62269186/150713653-c1457881-3a2a-4446-bb7f-8d8a25508eeb.gif)

**Comment by anatawa12** [comment]
I have the same problem.

It looks if there are no latin characters in checkbox, the GitHub will return HTTP 400/500 error.

It looks ids for checkbox (including POST request to GitHub) of the GitHub issue forms are generated based on all latin characters in the label.
In your & my case, the label doesn't have any latin characters. 
So it looks the id of checkbox is now empty string.
In your problem that you can't check some checkbox, because the 2~4th checkbox have same id `issue_form_checks_` so clicking 3rd and 4th label will trig 2nd checkbox.
Also, the GitHub looks disallow empty id for form field so creating issue with checkbox with empty id will return HTTP 400/500.

Currently, It looks there is no workaround except for adding some latin characters.
I think it's good to have some way to specify id for checkbox to avoid this so I've made a request: #11949

**Reply by AnotiaWang** [reply]
> I have the same problem.
> 
> It looks if there are no latin characters in checkbox, the GitHub will return HTTP 400/500 error.
> 
> It looks ids for checkbox (including POST request to GitHub) of the GitHub issue forms are generated based on all latin characters in the label. In your & my case, the label doesn't have any latin characters. So it looks the id of checkbox is now empty string. In your problem that you can't check some checkbox, because the 2~4th checkbox have same id `issue_form_checks_` so clicking 3rd and 4th label will trig 2nd checkbox. Also, the GitHub looks disallow empty id for form field so creating issue with checkbox with empty id will return HTTP 400/500.
> 
> Currently, It looks there is no workaround except for adding some latin characters. I think it's good to have some way to specify id for checkbox to avoid this so I've made a request: #11949

Thanks bro 👍
